polyglot/piranha/README.md

```diff
@@ -1,20 +1,171 @@
+    - [:computer: Command-line Interface](#computer-command-line-interface)
+        - [Get platform-specific binary from releases or build it from source following the below steps:](#get-platform-specific-binary-from-releases-or-build-it-from-source-following-the-below-steps)
+    - [Languages supported](#languages-supported)
+  - [Getting Started](#getting-started)
+    - [Demos](#demos)
+      - [Running the Demos](#running-the-demos)
+        - [Stale Feature Flag Cleanup:](#stale-feature-flag-cleanup)
+        - [Match-only rules:](#match-only-rules)
+        - [Structural Find/Replace](#structural-findreplace)
+        - [Structural Find/Replace with Custom Cleanup](#structural-findreplace-with-custom-cleanup)
+      - [Building upon the *stale feature flag cleanup* demo](#building-upon-the-stale-feature-flag-cleanup-demo)
+  - [*Stale Feature Flag Cleanup* in depth](#stale-feature-flag-cleanup-in-depth)
+    - [Adding support for a new feature flag system](#adding-support-for-a-new-feature-flag-system)
+      - [Adding a new API usage](#adding-a-new-api-usage)
+      - [Parameterizing the behavior of the feature flag API:](#parameterizing-the-behavior-of-the-feature-flag-api)
+    - [Adding Cleanup Rules](#adding-cleanup-rules)
+      - [Example](#example)
+  - [Piranha Arguments :](#piranha-arguments-)
+  - [Contributing](#contributing)
+    - [Naming conventions for the rules](#naming-conventions-for-the-rules)
+    - [Writing tests](#writing-tests)
+
+
+## Overview
+<p style="text-align:center;">
+<img src="images/piranha_architecture.svg" width="800" height="500" alt="Polyglot Piranha Architecture"/>
+</p>
+
+This is the higher level architecture of Polyglot Piranha. 
+At its heart, Polyglot Piranha is a structural find/replacement (rewrite) engine and pre-build language specific cleanup rules like - like simplifying boolean expressions, simplifying `if-else` statements, deleting empty class, deleting files with no type declarations, inline local variables, and many more. 
+A user provides :
+- A set (or, a graph) of structural find/replace rules 
+- Path to the code base
+- Arguments to modify Piranha's behavior (like deleting associated comments)
+When Piranha applies the set (or graph) of user defined rules, it trigger the __pre-built__ language specific cleanup rules to do a deep cleanup. 
+
+## When is Polyglot Piranha useful?
+#### Example 1 (Stale Feature Flag Cleanup)
+Let's take an example, where we know for a fact that the expression `exp.isTreated("SHOW_MENU")` always returns `true` (i.e. the feature *Show Menu* is treated)
+```
+public String fooBar(boolean x) {
+    if(exp.isTreated("SHOW_MENU")|| x){
+        String menu = getMenu();
+        return menu;
+    }
+    return "";
+}
+```
+To cleanup this code with Piranha, a user would have to *one* rule to update the expressions like `exp.isTreated("SHOW_MENU")` to `true` and hook it to the pre-built boolean simplification rules. It would result in : 
```

Comment:
"Would have to *one* rule" -> "Would have to **write** *one* rule" (I would even do "Would **only** have to **write** *one* rule" or maybe provide instead of write, your choice)